### DISCUSSION TITLE
Can Cabloy's frontend solution be used on its own?

### DISCUSSION THREAD

**Discussion by q807175022** [zhennann/cabloy#28] [Q&A]
I really like the author's mobile-first approach, which still provides a good experience on PC.
However, I'm a frontend developer and only have authority over frontend technical decisions. Is there a relatively convenient way to adopt Cabloy starting only from the frontend side?

Also, if it's convenient, I would appreciate it if the author could help answer a few questions:
In the current solution, it seems you implemented your own routing approach instead of using `vue-router` or `history.pushState`.
Because of this, when I instinctively use the browser's default Back button, I often end up leaving the entire system.

**Comment by zhennann** [comment]
抱歉这么晚看到你的问题，感谢你的认可，下面是我的一些思路与建议，欢迎继续交流：
1、Cabloy是前后端分离的架构设计，同时引入了模块化的架构体系。从你的角度，如果是保留现有后端技术栈的场景下，可以把Cabloy作为中台来引入
2、Cabloy采用了Framework7的router机制，在mobile场景下支持浏览器的history，在pc场景下不支持（因为后台管理系统更多是通过菜单+tab页签的机制来切换页面）
     * 而且，pc场景采用了pc=mobile+pad自适应机制，每一个页面都是一个单独的router容器，有独立的页面堆叠/回退的效果
     * 我在知乎有一篇回答，可以作为补充：[vue的keep-alive非常实用，为什么angular和react都不跟进呢？](https://www.zhihu.com/question/456149654/answer/1855148235)

<img width="617" alt="image" src="https://user-images.githubusercontent.com/24246985/233524778-94c5dcae-0af7-4ea2-91ac-3f5506eb37cc.png">
